### PR TITLE
Разделение больших чисел по разрядам

### PR DESCRIPTION
**Pull request by golmakov**
Опциональное правило разделяющее числа по разрядам

### CODE REVIEW

File: src/rules/ru/number/digitGrouping.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+Typograf.addRule({
+    name: 'ru/number/digitGrouping',
+    index: '310',
+    handler(text, settings) {
+        return text.replace(/(\d{5,}([.,]\d+)?)/g, function($0, $1) {
+            const delimiter = $1.match(/[.,]/);
+    
+            let parts = $1.split(delimiter);
```
Comment (hcodes) [inline src/rules/ru/number/digitGrouping.js:8]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split#Parameters
Первый параметр у split может быть строкой или регуляркой, а здесь null или массив.

File: src/rules/ru/number/digitGrouping.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+Typograf.addRule({
+    name: 'ru/number/digitGrouping',
+    index: '310',
+    handler(text, settings) {
+        return text.replace(/(\d{5,}([.,]\d+)?)/g, function($0, $1) {
+            const delimiter = $1.match(/[.,]/);
```
Comment (hcodes) [inline src/rules/ru/number/digitGrouping.js:6]:
```js
const delimiter = $1.match(/[.,]/);
if (!delimiter) {
    return $1;
}

const parts = $1.split(delimiter);

return parts[0].replace(/(\d)(?=(\d{3})+([^\d]|$))/g, '$1' + settings.space) +
    delimiter +
    parts[1];
```

File: src/rules/ru/number/digitGrouping.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+Typograf.addRule({
+    name: 'ru/number/digitGrouping',
```
Comment (hcodes) [inline src/rules/ru/number/digitGrouping.js:2]: Если удаляются или добавляются неразрывные пробелы, то группа `nbsp` (`ru/nbsp/digitGrouping`).

File: src/rules/ru/number/digitGrouping.js (hunk 1 above)
Comment (golmakov) [reply src/rules/ru/number/digitGrouping.js:8]: Спасибо, буду знать

**Comment by hcodes** [comment]
https://www.bipm.org/utils/common/pdf/si-brochure/SI-Brochure-9-EN.pdf

> 5.4.4    Formatting numbers, and the decimal marker The symbol used to separate the integral part of a number from its decimal part is called the decimal  marker.  Following  a  decision  by  the  22nd  CGPM  (2003,  Resolution  10),  the  decimal  marker  “shall  be  either  the  point  on  the  line  or  the  comma  on  the  line.”  The  decimal  marker  chosen  should  be  that  which  is  customary  in  the  language  and  context  concerned. If the number is between +1 and −1, then the decimal marker is always preceded by a zero. Following the 9th CGPM (1948, Resolution 7) and the 22nd CGPM (2003, Resolution 10), for numbers with many digits the digits may be divided into groups of three by a space, in order  to  facilitate  reading.  Neither  dots  nor  commas  are  inserted  in  the  spaces  between  groups  of  three.  However,  when  there  are  only  four  digits  before  or  after  the  decimal  marker, it is customary not to use a space to isolate a single digit. The practice of grouping digits  in  this  way  is  a  matter  of  choice;  it  is  not  always  followed  in  certain  specialized  applications such as engineering drawings, financial statements and scripts to be read by a computer.  For numbers in a table, the format used should not vary within one column.

**Comment by hcodes** [comment]
#350

**Comment by hcodes** [comment]
@golmakov 
Александр, спасибо за правило к Типографу.
Будет доступно в версии 6.10.0.

**Pull request closed without merging**